mlflow/deployments/interface.py

```diff
@@ -1,47 +1,55 @@
+import inspect
 from mlflow.deployments.plugin_manager import DeploymentPlugins
-from mlflow.utils import experimental
+from mlflow.deployments.base import BaseDeploymentClient
 
 
-listType = list
-plugin_store = DeploymentPlugins(auto_register=True)
+plugin_store = DeploymentPlugins()
 
 
-@experimental
 def get_deploy_client(target_uri):
     """
-    It calls the `get_deploy_client` method from the plugin and return an instance of a subclass of
-    :py:class:`mlflow.deployments.BaseDeploymentClient` that can be used to deploy models to the
-    specified target
+    It fetches all the classes inside the plugin and check for a subclass of
+    :py:class:`mlflow.deployments.BaseDeploymentClient` that can be used to deploy
+    models to the specified target
+
+    .. Note::
+        The plugin should only have one child class of
+        :py:class:`mlflow.deployments.BaseDeploymentClient` else it throws.
 
     :param: target_uri: URI of target to deploy to. Run ``mlflow deployments --help`` via the CLI
                         for more information on supported deployment targets
     """
-    # TODO: Maybe we should keep the separator as ":/" instead of "://"?
     target = target_uri.split("://")[0]
```

Comment:
As commented above, can we use logic like that in the [RunsArtifactRepository](https://github.com/mlflow/mlflow/blob/07472fe6fe3b162ff1afbd75969c3b0b35851ee0/mlflow/store/artifact/runs_artifact_repo.py#L36-L50) to verify that the target URI is a valid URI before trying to index into the plugin store?

Reply:
Sounds good